### PR TITLE
Execute a single command where single quotes are needed.

### PR DESCRIPTION
**Pull request by jrswab**
This is in response to issue #32 where the user wanted to enter a command where they specified the shell.

eg. `bash -c 'echo testing'`.

This fix will not handle a command with trailing `&&` at this time but after looking at the readme it seems that may not be the proper use case anyway.

eg. `bash -c 'echo testing' && echo again`

### CODE REVIEW

**Comment by bitfield** [comment]
Thanks @jrswab! Very nice PR.

Does this fix issue #32 completely? It's not completely clear to me from looking at it whether or not it does.

Could we add a test case which demonstrates the problem reported in #32, which fails with the current version of `script`, but which your patch makes pass?

If this doesn't correctly handle commands with `&&`, is that a new issue that needs to be raised, or should we consider it part of issue #32 and fix it in a separate PR?

**Comment by jrswab** [comment]
> Does this fix issue #32 completely?
It does fix the user's problem completely as well as any other command like it.

> Could we add a test case which demonstrates the problem reported in #32, which fails with the current version of script, but which your patch makes pass?
Sure, do you have a desired way of testing? I was in `filters_test.go` before the PR looking it over. Not sure if you would want the test to be apart of that or `sources_test.go`

> If this doesn't correctly handle commands with &&, is that a new issue that needs to be raised, or should we consider it part of issue #32 and fix it in a separate PR?
If you want the users to have the ability of multiple commands after the initial command like the the one in issue #32 then I'd say it's a new PR. Mainly because that user's direct issue is solved with the code update.

If you want I can step through the code as a comment here and explain since it's not very long.

**Comment by bitfield** [comment]
I'm a bit confused that the issue is about spaces and you're referring to quotes in this PR. It doesn't look as though your patch actually deals with spaces, or am I wrong?

The substantive implementation of Exec is as a filter, so yes, I would add a set of test cases to `TestExecFilter`, along the lines of `TestBasename` in the same file, with various combinations of spaces, quotes, and other edge cases.

**Comment by jrswab** [comment]
> I'm a bit confused that the issue is about spaces and you're referring to quotes in this PR. It doesn't look as though your patch actually deals with spaces, or am I wrong?

Yeah, this is a bit confusing. When I hug into the issue I noticed the code already splits the command on the spaces. The reason why his example was not working was not due to the space but the `'`. When `strings` breaks up the command we end up with `bash` `-c` `'echo` `'test` but in order for the shell to execute `'echo test'` needs to be one parameter.

> The substantive implementation of Exec is as a filter, so yes, I would add a set of test cases to TestExecFilter, along the lines of TestBasename in the same file, with various combinations of spaces, quotes, and other edge cases.

Ok I'll do that this weekend when I get some free time.

**Comment by bitfield** [comment]
Useful comment on #32 which clarifies what we need here:

> Quoting does not help: The input string is passed to [`strings.Field`](https://github.com/bitfield/script/blob/master/filters.go#L116), which is passed to the `exec.Command` in the next line.
> 
> Unfortunately, the strings.Fields does not group into a single field quoted words:
> 
> ```go
> fmt.Println(strings.Join(strings.Fields(`a b "c d" e`), ", "))
> // Output: a, b, "c, d", e
> ```
> 
> Notice that in this example, `"c` is one string returned from `strings.Fields` and `d"` is another, while for the `command.Exec` you would want `c d` as a single string.

**Comment by jrswab** [comment]
> `fmt.Println(strings.Join(strings.Fields('a b "c d" e'), ", "))// Output: a, b, "c, d", e`
>
> Notice that in this example, `"c` is one string returned from `strings.Fields` and `d"` is another, while for the `command.Exec` you would want `c d` as a single string.

Let me set up those tests and add this as an example then tweak the code as needed. It seems to be the same (or a similar) use case that I was aiming to fix with the submitted code.

**Comment by jrswab** [comment]
Just an update:
I am rewriting this to handle a few more cases that will be included in the test. After a bit more thinking I want to make sure it handles more weird cases with quotes than just what #32 was referencing.

**Comment by bitfield** [comment]
Sounds good @jrswab! Let me know if you want another pair of eyes on it.

**Comment by bitfield** [comment]
How are you getting along @jrswab? Want any help?

**Comment by jrswab** [comment]
> How are you getting along @jrswab <https://github.com/jrswab>? Want any help?

I got tied up with life recently. If you want we to push up what I have currently I'd be happy to. That way anyone can help out with it.

**Comment by bitfield** [comment]
Certainly! That would be very helpful, thank you.

File: filters_test.go [hunk 1]
```diff
@@ -189,6 +189,27 @@ func TestExecFilter(t *testing.T) {
...
 		t.Error("want exec on erroneous pipe to be a no-op, but it wasn't")
 	}
+
+	// Testing cases with double quotes wrapped in single quoutes.
+	testPipe := NewPipe()
+	testQuotes := []struct {
+		testStrings string
+		want        string
+	}{
+		{`echo 'test'`, "test"},
```
Comment (bitfield) [inline filters_test.go]: Won't this only work if the `echo` command exists and is in the user's path? Ideally, we wouldn't need to depend on any userland tools for `script` to work; that's part of the reason for writing it. If we just want to test that we correctly assemble command line arguments for a given command, we don't need to actually run the command to do that, do we?

**Comment by bitfield** [comment]
Thanks for your work on this, @jrswab! With your help I've fixed this issue in v0.14.2 via #42.
